Packages/io.chainsafe.web3-unity/Runtime/Scripts/WalletConnect/Dialog/LocalWalletButton.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Threading.Tasks;
+using ChainSafe.Gaming.WalletConnect.Models;
+using TMPro;
+using UnityEngine;
+using UnityEngine.Networking;
+using UnityEngine.UI;
+
+namespace ChainSafe.Gaming.WalletConnect.Dialog
+{
+    public class LocalWalletButton : MonoBehaviour
+    {
+        public GameObject IconStub;
+        public Image Icon;
+        public TMP_Text Text;
+        public Button Button;
+
+        public async void Set(WalletModel data, Action onClick)
+        {
+            // Text
+            Text.text = data.Name;
+            
+            // Click
+            Button.onClick.RemoveAllListeners();
+            Button.onClick.AddListener(() => onClick());
+            
+            // Icon
+            IconStub.gameObject.SetActive(true);
+            Icon.gameObject.SetActive(false);
+
+            if (data.Images == null || string.IsNullOrEmpty(data.Images.SmallUrl))
+            {
+                return;
+            }
+            
+            var icon = await DownloadIcon(data.Images.SmallUrl);
+            if (icon == null)
+            {
+                return;
+            }
+            
+            Icon.sprite = icon;
+            Icon.gameObject.SetActive(true);
+            IconStub.gameObject.SetActive(false);
+        }
+
+        private static async Task<Sprite> DownloadIcon(string url)
```

Comment:
why not?

Reply:
because it's private and it has only one usage where it doesn't need to be static

Reply:
Making it static is the optimization that the IDE suggests when the method doesn't use any instance members. How would this be better if we make it non-static?

Reply:
http://wiki.c2.com/?GlobalVariablesAreBad

Reply:
Global Vars are bad 100%, but static methods, I think, are getting optimized by the compiler because they for sure don't access any instance members.
`public static int PlayerHealth;` is not the same as `private static NotAccessedAnywhereElseButOnlyInThisClass()`.

Reply:
I know but form a design perspective there's no reason to make a method static unless we need to, especially a private static method which basically has zero value in a non-static context/class, we can make it public but it's still out of context since `LocalWalletButton` isn't really where a globally accessible `DownloadIcon` should exist, maybe in a Utils class of sorts.

Reply:
[This](https://stackoverflow.com/questions/3151768/are-global-static-classes-and-methods-bad) is a good discussion on the issue

Reply:
resolving based on feedback on architecture meeting